### ISSUE TITLE
Handler Exceptions settings

### ISSUE THREAD

**Issue by BJury** [OrchardCMS/OrchardCore#5941]
I have a class that implements ContentPartDisplayDriver<MyPart>, and DisplayAsync(MyPart part, BuildPartDisplayContext) throws an exception.

When I call '@(await DisplayAsync(Model.Content.CommodityPermissionsPart))' in a view, the exception gets ignored and the part is not displayed. (No redirect to an error page or similar.)

There is a message in the Debug output but that is all.

**Comment by deanmarcussen** [comment]
To clarify are you intentionally throwing the exception and wanting an error page?

**Comment by BJury** [comment]
I wasn't intentionally throwing it no.

I was using it as a way to enable permissions based on a taxonomy, so if it silently fails its the worst result.

**Comment by deanmarcussen** [comment]
What I meant was are you using the exception to try and manage permissions, by throwing when it doesn't meet the requirements?

The Content Drivers do all try catch, and log without rethrowing exceptions, unless the exception is considered Fatal (OutOfMemoryException, SecurityException, or SEHException)

However if you're doing this as a permissions / authorization then it might be better handled by implementing an `IAuthorizationHandler` (like the `ContentTypeAuthorizationHandler`) which would just call `ContentItem.As<YourPart>()` and then if the part exists (i.e. is not null) succeed or fail it based on your logic.

Maybe I am misunderstanding what you are trying to achieve though?

**Comment by BJury** [comment]
No, it works in a similar way to [this](https://github.com/EtchUK/Etch.OrchardCore.ContentPermissions/blob/master/Drivers/ContentPermissionsDisplay.cs). It redirects to /error/403 on a permissions fail.

> The Content Drivers do all try catch, and log without rethrowing exceptions, unless the exception is considered Fatal

So exceptions are just consumed, and there is no way to tell if something has failed? Its weird as an exception in the view itself does something else.

**Comment by deanmarcussen** [comment]
> No, it works in a similar way to [this](https://github.com/EtchUK/Etch.OrchardCore.ContentPermissions/blob/master/Drivers/ContentPermissionsDisplay.cs). It redirects to /error/403 on a permissions fail.

Ok, that will work, `IAuthorizationHandler` would just be my suggestion.

> So exceptions are just consumed, and there is no way to tell if something has failed? Its weird as an exception in the view itself does something else.

Suggest you place your own try catch block in the driver maybe?

**Comment by BJury** [comment]
> Suggest you place your own try catch block in the driver maybe?
For sure.

The point what would you expect a exception when calling DisplayAsync<>() to do? Personally I'd expect to see an error message, rather than it do nothing.

**Comment by sebastienros** [comment]
Maybe alter the current response status code?

**Comment by BJury** [comment]
My point isn't how to achieve what I am doing permissions wise, or handle a error.

Its that unhandled exceptions that are thrown when calling DisplayAsync() get silently consumed. While exceptions in the view, or elsewhere, would result in the error page being shown.

I'd have though this was an important issue.
